File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.rs
impl Solution {
pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
    let mut output = Vec::new();
    let mut currentSum = 0;

        for &i in nums.iter() {
            currentSum = currentSum + i;
            output.push(currentSum);
        }
    return output;    
    }
}

File: 1603-running-sum-of-1d-array/running-sum-of-1d-array.java
class Solution {
    public int[] runningSum(int[] nums) {
        
        int[] output = new int[nums.length];
        int currentSum = 0;
        for(int i = 0; i < nums.length; i++) {
            
            output[i] = currentSum + nums[i];
            currentSum += nums[i];
        }

        return output;
    }
}